Missing result type for shift

The following C
```c
unsigned foo(void) {
    unsigned char shift = 0;
    return (unsigned)(5) << shift;
}
```

Produces this zig:

```zig
pub export fn foo() c_uint {
    var shift: u8 = 0;
    _ = &shift;
    return @as(c_uint, @bitCast(@as(c_int, @as(c_int, 5)))) << @intCast(@bitCast(@as(c_int, @as(c_int, shift))));
}
```

Which fails to compile:
```
test.zig:7:73: error: @bitCast must have a known result type
    return @as(c_uint, @bitCast(@as(c_int, @as(c_int, 5)))) << @intCast(@bitCast(@as(c_int, @as(c_int, shift))));
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test.zig:7:73: note: use @as to provide explicit result type
```

Removing the cast, changing `shift` to `unsigned`, or using a compile-time constant instead of variable shift makes the problem go away.

--- a/src/Translator.zig
+++ b/src/Translator.zig
@@ -2677,7 +2677,7 @@ fn transShiftExpr(t: *Translator, scope: *Scope, bin: Node.Binary, op_id: ZigTag
     // lhs >> @intCast(rh)
     const lhs = try t.transExpr(scope, bin.lhs, .used);
 
-    const rhs = try t.transExprCoercing(scope, bin.rhs, .used);
+    const rhs = try t.transExpr(scope, bin.rhs, .used);
     const rhs_casted = try ZigTag.int_cast.create(t.arena, rhs);
 
     return t.createBinOpNode(op_id, lhs, rhs_casted);

--- a/test/cases/translate/bitshift.c
+++ b/test/cases/translate/bitshift.c
@@ -5,5 +5,5 @@ int foo(void) {
 // translate
 //
 // pub export fn foo() c_int {
-//     return (@as(c_int, 1) << @intCast(2)) >> @intCast(1);
+//     return (@as(c_int, 1) << @intCast(@as(c_int, 2))) >> @intCast(@as(c_int, 1));
 // }

--- a/test/cases/translate/bitshift_with_bitcast.c
+++ b/test/cases/translate/bitshift_with_bitcast.c
@@ -1,0 +1,9 @@
+unsigned int bit31() {
+    return 1u << (31);
+}
+
+// translate
+//
+// pub export fn bit31() c_uint {
+//     return @as(c_uint, 1) << @intCast(@as(c_uint, @bitCast(@as(c_int, @as(c_int, 31)))));
+// }

